docs/Versions.md

```diff
@@ -17,7 +17,7 @@ Versioned formulae we include in [homebrew/core](https://github.com/homebrew/hom
 * Submitted versioned formulae should be expected to be used by a large number of people. If this ceases to be the case, they will be removed. We will aim not to remove those in the [top 3,000 `install_on_request` formulae](https://brew.sh/analytics/install-on-request/).
 * Versioned formulae should not have `resource`s that require security updates. For example, a `node@6` formula should not have an `npm` resource but instead rely on the `npm` provided by the upstream tarball.
 * Versioned formulae should be as similar as possible and sensible compared to the main formulae. Creating or updating a versioned formula should be a chance to ask questions of the main formula and vice versa, e.g. can some unused or useless options be removed or made default?
-* No more than five versions of a formula (including the main one) will be supported at any given time, regardless of usage. When removing formulae that violate this, we will aim to do so based on usage and support status rather than age.
+* No more than five versions of a formula (including the main one) will be supported at any given time, unless they have over 1000 [analytics 30 days installs](https://formulae.brew.sh/analytics/install/30d/) of usage. When removing formulae that violate this, we will aim to do so based on usage and support status rather than age.
```

Comment:
Do we want to specifically call out that this is for compilers? I'm not currently able to think of anything else that would have at least 5 versions at once in homebrew-core.

Reply:
I'd rather make it more general. This was added because there was initially a bit of a maintainer uproar about having any versions at all. Given how we've seen it pan out, though: I can't see why a widely used older version of anything should be removed when we're not removing similar unversioned software. More versions == more useful to Homebrew users.

Reply:
Should we specify that they should also still match our rules? Or are packages going to stay around forever as long as they are popular? Because in that case we'll have to keep a lot of ancient php versions

Reply:
> Because in that case we'll have to keep a lot of ancient php versions

This is a great example, and kind of why I floated calling out compilers. The risk of using `llvm@11` is much different than running `php@7.4`. One thing that I love that we do, despite user complaints, is essentially act as a forcing function to move people off of unsupported versions. When talking user experience + security, I think we can all agree the most common thing is "I won't upgrade until I'm forced to".

Reply:
> Because in that case we'll have to keep a lot of ancient php versions

If they don't have CVEs and are supported by upstream (which is in another bullet): yeh, we probably should do this.

Reply:
The supported by upstream is what saves us here with PHP, and is how we are dealing with it now. The bigger question is how are we deciding if compilers are "supported" upstream? The last release of `llvm@12` was in July 2021, and we have not deprecated it yet. This is why I keep circling back to compilers being primarily affected by this.

Reply:
@p-linnane I feel like to me support isn't something that's inferred as much as explicitly stated. If they call it "EOL" or whatever: there's no support. If they don't state explicitly: judge it based on usage instead.

Reply:
I suppose I can be on board with that. At the end of the day, it's about better supporting our users. As long as we don't accept new formulae that depend on old versions, it should be fine until we can't bottle them anymore. (`llvm@12` is a good example for this as well, since we have not been able to bottle it on Sonoma).